crates/ty_server/src/server/api/notifications/did_change_watched_files.rs

```diff
@@ -53,6 +57,13 @@ impl SyncNotificationHandler for DidChangeWatchedFiles {
                 continue;
             };
 
+            if matches!(
+                system_path.file_name(),
+                Some(".gitignore" | ".ignore" | "ty.toml" | "pyproject.toml")
+            ) {
+                project_changed = true;
+            }
+
```

Comment:
I'd prefer if we put this into `apply_changes` to avoid duplicating the logic when a project has changed.

Reply:
I think this change already exists in `apply_changes`

https://github.com/astral-sh/ruff/blob/cd1fd80f595b3e41943d11f8479b34460d446bd1/crates/ty_project/src/db/changes.rs#L45-L53

, this is present here mainly to check whether we need to refresh the diagnostics or not as the `didChangeWatchedFiles` is registered for `.py`, `.pyi`, and `.ipynb` files as well for which we shouldn't need to do that.

Reply:
Yes. I'd prefer if we wouldn't duplicate the logic here and instead add a return value to `apply_changes` indicating what changed

Reply:
Got it, I've made the change to return a `ChangeResult` that contains boolean fields to indicate what changes were detected by the `apply_changes` method.